Realm/ObjectServerTests/RLMPermissionsAPITests.m

```diff
@@ -43,7 +43,7 @@
 
 #define CHECK_PERMISSION_COUNT(ma_results, ma_count) CHECK_PERMISSION_COUNT_PREDICATE(ma_results, ma_count, ==)
 
-#define CHECK_PERMISSION_PRESENT(ma_results, ma_permission) {                                                          \
+#define CHECK_PERMISSION_PRESENT(ma_results, ma_permission) @autoreleasepool {                                                          \
```

Comment:
Which thing is this trying to avoid having escape from the block? Now that I'm looking at it again, the whole thing is a bit suspicious: the notification block should be using the RLMResults passed to it rather than holding onto a weak reference (which would happen to eliminate the autorelease from resolving the weak reference), and it should be calling `-invalidate` on the token rather than using `objc_precise_lifetime` to better match what actual users would be doing.

Reply:
I was originally troubleshooting a failed assertion somewhere deep inside XCTest code that was logged to console, but did not cause the offending test to crash or fail. The issues you brought up are problematic indeed; I'm thinking about just replacing all these macros with functions that take line number/file name arguments. (I added the weak reference originally because otherwise the compiler emitted a warning about a potential retain cycle, something that doesn't seem to happen in similar non-macro code.)